### PR TITLE
[cronet_http] Upgrade jni to 0.9.2 and publish 1.2.1

### PR DESCRIPTION
**Pull request by HosseinYousefi**
Fix our users' issue described here: https://github.com/dart-lang/native/issues/1119.

@brianquinlan Please manually publish as cronet currently needs Flutter and it won't work with the autopublish CI.

### CODE REVIEW

File: pkgs/cronet_http/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+## 1.2.1
```
Comment (brianquinlan) [inline pkgs/cronet_http/CHANGELOG.md:1]: Please change this to "1.2.1-wip"

File: pkgs/cronet_http/pubspec.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name: cronet_http
-version: 1.2.0
+version: 1.2.1
```
Comment (brianquinlan) [inline pkgs/cronet_http/pubspec.yaml:2]: Please change this to "1.2.1-wip"

**Review by brianquinlan — CHANGES_REQUESTED** [review]
The tests timed-out so I restarted them. I have one more Dart 3.4-releated change to land and then I will publish this.

**Comment by brianquinlan** [comment]
The emulator isn't even starting before the tests time out. Hopefully this is a temporary github glitch.

**Comment by techouse** [comment]
This happens with AGP 8.4.0. Works fine with AGP 8.3.2

**Comment by techouse** [comment]
Not on the app level. It has most likely something to do with the jni plugin.

**Comment by brianquinlan** [comment]
I think that the CI is broken. Working on fixing it.

**Review by brianquinlan — APPROVED** [review]

**Comment by HosseinYousefi** [comment]
> Not on the app level. My
> 
> ```groovy
> android.defaultConfig.minSdkVersion 24
> ```
> 
> still fails with the same error.
> 
> It has most likely something to do with the jni plugin.

I meant after the fix to jni. You can override the jni dependency to use 0.9.2 and try.

**Comment by HosseinYousefi** [comment]
> I think that the CI is broken. Working on fixing it.

Thanks! Merge this whenever you see fit then.

**Comment by techouse** [comment]
> I meant after the fix to jni. You can override the jni dependency to use 0.9.2 and try.

You mean just use this PR instead of `v1.2.0`? Sure, gimme a few minutes.

**Comment by HosseinYousefi** [comment]
> > I meant after the fix to jni. You can override the jni dependency to use 0.9.2 and try.
> 
> You mean just use this PR instead of `v1.2.0`? Sure, gimme a few minutes.

Yeah, you can do that too, 

Here's the dependency!
```yaml
cronet_http:
  git:
    url: https://github.com/dart-lang/http
    ref: jni-0.9.2
    path: pkgs/cronet_http
```

**Comment by techouse** [comment]
Yep, compiles ok using

```yaml
dependency_overrides:
  cronet_http:
    git:
      url: https://github.com/dart-lang/http.git
      ref: jni-0.9.2
      path: pkgs/cronet_http
```

and `android/app/build.gradle`

```groovy
dependencies {
  implementation(platform("org.jetbrains.kotlin:kotlin-bom:1.9.24"))
}
```

It only complains about
```
/Users/techouse/.pub-cache/hosted/pub.dev/jni-0.9.2/android/src/main/java/com/github/dart_lang/jni/JniPlugin.java uses or overrides a deprecated API.
```

**Comment by HosseinYousefi** [comment]
> It only complains about
> 
> ```
> /Users/techouse/.pub-cache/hosted/pub.dev/jni-0.9.2/android/src/main/java/com/github/dart_lang/jni/JniPlugin.java uses or overrides a deprecated API.
> ```

Yeah, that's not a problem. Although, I thought I have silenced this warning here: https://github.com/dart-lang/native/blob/main/pkgs/jni/android/src/main/java/com/github/dart_lang/jni/JniPlugin.java#L22 🤔

**Pull request merged**